src/algebra/homology/local_cohomology.lean

```diff
@@ -87,15 +87,26 @@ of `R`, represented as a functor `I`. -/
 In this definition we do not assume any special property of the diagram `I`, but the relevant case
 will be where `I` is (cofinal with) the diagram of powers of a single given ideal.
 
-Below, we give two equivalent (to be shown) definitions of the usual local cohomology with support
+Below, we give two equivalent definitions of the usual local cohomology with support
 in an ideal `J`, `local_cohomology` and `local_cohomology.of_self_le_radical`.
 
-TODO: Show that any functor cofinal with `I` gives the same result.
  -/
 def of_diagram (I : D ⥤ ideal R) (i : ℕ) :
   Module.{max u v} R ⥤ Module.{max u v} R :=
 colimit (diagram.{(max u v) v} I i)
 
+variables {E : Type v} [small_category E]
+  (I' : E ⥤ D) [functor.initial I']
+  (I : D ⥤ ideal R)
+
+def diagram_comp (i : ℕ) : diagram (I' ⋙ I) i ≅ I'.op ⋙ (diagram I i) := iso.refl _
+
+/-- Local cohomology agrees along cofinal diagrams. -/
+def iso_of_final (i : ℕ) :
```

Comment:
The functor here involves `I'.op`, which is why we use `functor.final.colimit_iso`. Implicitly we're getting finality from `functor.final_op_of_initial` (or maybe `functor.final_of_initial_op`). Should this explained in the comments here?